draft-ietf-tls-esni.md

```diff
@@ -174,14 +174,31 @@ ClientHelloInner in an "encrypted_client_hello" extension, which this document
 defines ({{encrypted-client-hello}}). Finally, it sends ClientHelloOuter to the
 server.
 
-Upon receiving ClientHelloOuter, the client-facing server processes the
-encrypted ClientHelloInner and either terminates the connection (in Shared Mode)
-or forwards it to the backend server (in Split Mode).
+Upon receiving the ClientHelloOuter, the client-facing server takes one of the
+following actions:
+
+1. If it does not support ECH, it ignores the "encrypted_client_hello" extension
+   and proceeds with the handshake as usual, per {{RFC8446}}, Section 4.1.2.
+1. If it supports ECH but cannot decrypt it, then it ignores the extension and
+   proceeds with the handshake as usual. This is referred to as "ECH rejection".
+   When ECH is rejected, the server sends an acceptable ECH configuration in its
+   EncryptedExtensions message.
+1. If it supports ECH and can decrypt it, then it forwards the CllientHelloInner
+   to the backend, who terminates the connection. This is referred to as "ECH
+   acceptance".
+
+[[TODO: Ensure this is specified below: What does the server do if it can
+decrypt but decryption fails, i.e., the ciphertext is invalid? Does it abort or
+proceed with the outer CH?]]
+
+Upon receiving the server's response, the client determines whether ECH was
+accepted or rejected and proceeds with the handshake accordingly. (See
+{{client-behavior}}.)
```

Comment:
```suggestion
{{client-behavior}} for details.)
```

Reply:
Oops, missed this. Shall I make this change/